core/src/main/java/com/datastax/oss/driver/internal/core/session/PoolManager.java

```diff
@@ -314,10 +314,14 @@ private void processStateEvent(NodeStateEvent event) {
       NodeState newState = event.newState;
       if (pending.containsKey(node)) {
         pendingStateEvents.put(node, event);
-      } else if (newState == NodeState.FORCED_DOWN) {
+      } else if (newState == null || newState == NodeState.FORCED_DOWN) {
         ChannelPool pool = pools.remove(node);
         if (pool != null) {
-          LOG.debug("[{}] {} was FORCED_DOWN, destroying pool", logPrefix, node);
+          LOG.debug(
+              "[{}] {} was {}, destroying pool",
+              logPrefix,
+              node,
+              newState == null ? "removed" : "forced down");
```

Comment:
total nit here: Does it matter much that the debug log changes `FORCED_DOWN` to `forced down` in those cases? I realize it's just a log, and a Debug log at that. Again, total nitpick.

Reply:
You're right, debug logs don't matter much but there's no reason to change it either. If anything, using the constant name gives you a hint of what to look for in the source code.
I'll change it back.